src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs

```diff
@@ -161,9 +161,8 @@ private void PerformPackageUpdatesOrReinstalls()
         {
             try
             {
-                await PreviewAndExecuteUpdateActionsforSinglePackage();
-
-                if (!_isPackageInstalled)
+                var isPackageInstalled = await PreviewAndExecuteUpdateActionsforSinglePackage();
```

Comment:
`IsPackageInstalled(...)` returning bool is natural, and call sites read very naturally for it.

In contrast, it doesn't make sense that `PreviewAndExecuteUpdateActionsforSinglePackage()` returns bool; one has to read its implementation to understand what its return value represents.  This problem is solved by moving the call to `IsPackageInstalled(...)` to here, like so:

    var isPackageInstalled = await IsPackageInstalled(Id);
    await PreviewAndExecuteUpdateActionsForSinglePackage();